Move away from `cached_download`

Wauplin opened the pull request:
Follow-up PR after https://github.com/huggingface/diffusers/pull/8417.

In #8417, we introduce a mirror of the community pipeline modules as a HF dataset. This means we can now download modules from HF using `hf_hub_download` directly.

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/diffusers/pr_8419). All of your documentation changes will be reflected on that endpoint. The docs are available until 30 days after the last update.

sayakpaul commented on src/diffusers/utils/dynamic_modules_utils.py:
```diff
@@ -282,16 +277,15 @@ def get_cached_module_file(
-            resolved_module_file = cached_download(
-                github_url,
+            resolved_module_file = hf_hub_download(
+                repo_id="diffusers/community-pipelines-mirror",
```
(nit): `repo_id` could be a constant here.

sayakpaul reviewed (commented):
This design looks good to me!

Wauplin commented:
@sayakpaul I've addressed your comment + added a custom error to ask to open an issue if a revision is not correctly pushed. Hopefully it'll be fine for 99.9% of users using `main` or a recent-enough tag. Let me know what you think.

sayakpaul commented:
Killer work here!
